apps/openmw/mwgui/inventorywindow.cpp

```diff
@@ -528,14 +525,26 @@ namespace MWGui
         if (mDragAndDrop->mIsOnDragAndDrop)
         {
             MWWorld::Ptr ptr = mDragAndDrop->mItem.mBase;
+            int itemType = ptr.getContainerStore()->getType(ptr);
+
             mDragAndDrop->finish();
 
             if (mDragAndDrop->mSourceModel != mTradeModel)
             {
                 // Move item to the player's inventory
                 ptr = mDragAndDrop->mSourceModel->moveItem(mDragAndDrop->mItem, mDragAndDrop->mDraggedCount, mTradeModel);
             }
+
             useItem(ptr);
+
+            // If item is ingredient or potion don't stop drag and drop to simplify action of taking more than one 1 item
```

Comment:
How about a more generic implementation: if after `useItem` the item's count decreased by one and is greater than zero, restore the drag action. This way, the code doesn't need to be concerned with what is an ingredient or potion and if new actions/item types are added in the future then this will work for those too.

Reply:
The question is if we want to have this behaviour for other item types as well. Here: http://bugs.openmw.org/issues/4082 the author of the topic mentioned only ingredients and potions in the topic and last comment. I also think such behaviour for stacked clothes for example would be pointless.

Reply:
> I also think such behaviour for stacked clothes for example would be pointless.

Hm, true. Go ahead then.